api/utils/sshutils/conn.go

```diff
@@ -105,3 +118,57 @@ func (d *DialReq) CheckAndSetDefaults() error {
 	}
 	return nil
 }
+
+func (d *DialReq) MarshalToJSON() ([]byte, error) {
+	if d == nil {
+		return nil, trace.BadParameter("DialReq is 'nil'")
+	}
+
+	// if `ClientSrcAddr` or `ClientDstAddr` are not `net.TCPAddr` we need to construct it manually so it
+	// could be unmarshalled later correctly
+	getNetTCPAddr := func(addr net.Addr) *net.TCPAddr {
+		if addr == nil {
+			return nil
+		}
+
+		if tcpAddr, ok := addr.(*net.TCPAddr); ok {
+			return tcpAddr
+		}
+
+		host, portString, err := net.SplitHostPort(addr.String())
+		if err != nil {
+			return nil
+		}
+		portNumber, err := strconv.Atoi(portString)
+		if err != nil {
+			return nil
+		}
+
+		ip := net.ParseIP(host)
+		if ip == nil {
+			return nil
+		}
+
+		return &net.TCPAddr{
+			IP:   ip,
+			Port: portNumber,
+		}
+	}
+
+	clone := *d
+	clone.ClientSrcAddr = getNetTCPAddr(clone.ClientSrcAddr)
+	if getNetTCPAddr(clone.ClientSrcAddr) == nil {
+		clone.ClientSrcAddr = nil
+	}
+	clone.ClientDstAddr = getNetTCPAddr(clone.ClientDstAddr)
+	if getNetTCPAddr(clone.ClientDstAddr) == nil {
+		clone.ClientDstAddr = nil
+	}
```

Comment:
Leftover code?
```suggestion
	clone := *d
	clone.ClientSrcAddr = getNetTCPAddr(clone.ClientSrcAddr)
	clone.ClientDstAddr = getNetTCPAddr(clone.ClientDstAddr)
```

Reply:
It was to get clean nil instead of typed nil, but the whole thing is not needed now.